### PR TITLE
Add shfmt

### PR DESCRIPTION
**Pull request by elken**
Add support for shfmt. _Hopefully_ there's a neater way to do the installer, I'm not a gopher.

### CODE REVIEW

File: apheleia.el [hunk 1]
```diff
@@ -922,6 +922,7 @@ being run, for diagnostic purposes."
     (ocamlformat . ("ocamlformat" "-" "--name" filepath
                     "--enable-outside-detected-project"))
     (prettier . (npx "prettier" "--stdin-filepath" filepath))
+    (shfmt . ("shfmt" "-ci"))
```
Comment (raxod502) [inline apheleia.el:925]: Should we use the default settings by default, and let people customize it if they want to add additional options? Or do you think most people would want `-ci` so we should include it by default?

File: test/formatters/apheleia-ft.el [hunk 2]
```diff
@@ -260,7 +260,7 @@ environment variable, defaulting to all formatters."
                  in-file)
                (list stdout-buffer stderr-file)
                nil
-               (cdr command)))
+               (cdr `(,command))))
```
Comment (raxod502) [inline test/formatters/apheleia-ft.el:263]: Isn't the `cdr` of a list with one element always going to be nil?

**Review by raxod502 — COMMENTED** [review]
Thanks! This looks good, I pushed a commit that simplifies the installation (in my experience, Go tooling is atrocious and it's almost always better to just download a separate binary release if one is available), and I think all we need is a changelog entry.

File: apheleia.el (hunk 1 above)
Comment (elken) [reply apheleia.el:925]: Yeah you're right here, we shouldn't make decisions by default.

File: test/formatters/apheleia-ft.el (hunk 2 above)
Comment (elken) [reply test/formatters/apheleia-ft.el:263]: Hm yes you're right, this is a leftover change from a previous impl (I tried to use `sh-shell` and `indent-tabs-mode`) so I can remove it now I'm using a flat list

**Comment by elken** [comment]
> I pushed a commit that simplifies the installation

The issue with that is it doesn't take into consideration other CPU archs (which is why I opted to go for the build in the first place)

**Comment by raxod502** [comment]
Upstream offers binary downloads in all of the architectures that your switch statement covers; does the latest commit work for you?

**Comment by elken** [comment]
`latest_release` doesn't run for me, and I'm also not on linux :p So I need the `uname` snippet

**Comment by raxod502** [comment]
Those scripts are intended to be run inside Docker as part of the Apheleia build system (`make fmt-build FORMATTERS=shfmt`). Are you running them manually?

**Comment by elken** [comment]
I didn't think so, but I'll check again later

(Otherwise this is probably fine to merge)

**Comment by raxod502** [comment]
Installation succeeds and tests for shfmt are passing in CI at https://github.com/radian-software/apheleia/runs/6397390532, so I'll merge.

**Pull request merged**